base/logging.jl

```diff
@@ -492,8 +492,10 @@ end
 
 LogState(logger) = LogState(LogLevel(_invoked_min_enabled_level(logger)), logger)
 
+const CURRENT_LOGSTATE = ScopedValue{Union{Nothing, LogState}}(nothing)
+
 function current_logstate()
-    logstate = current_task().logstate
+    logstate = CURRENT_LOGSTATE[]
     return (logstate !== nothing ? logstate : _global_logstate)::LogState
```

Comment:
I suspect this will be a rather common pattern (i.e. use the scoped value, or fall back on to the global)
```suggestion
    return something(logstate, _global_logstate)::LogState
```